### PR TITLE
Added status field to m.notice.

### PR DESCRIPTION
**Pull request by Half-Shot**
Most of the time, notices are used by bots to respond to a user with either a failure or a success. "This is the gif we found" or "we couldn't find a gif called 'dog with a jetpack'". Right now bots all have their own phrasing for this whereas it could be part of the metadata of an event to include the status of an operation. Representing this on the client could be as simple as adding a warning triangle next to a message, or highlighting it in a sensible colour that works with the theme.

Problems:
* I came up with a few status options, but I'm not sure if that covers all of them, or too many.
* Should they start with ``m.`` or kept as is. That depends if custom statuses are a thing. I'd prefer them not to be, because that defeats the point of having statuses at all.

Example of how I expect it to look, in a messy CSS way.
errornotice

### CODE REVIEW

**Comment by richvdh** [comment]
ok to test

**Comment by richvdh** [comment]
@ara4n says:

> hum. i'm worried that it collides with the idea of richer events in general
and a specific solution for notices may miss the bigger picture
but i need to remember what we converged on for handling errors from bots
which is basically what this is
from wherever that discussion was we even did a use case for "loglevel" style events
ie "this is a net.arasphere.temperature event, with value 60C, and a _display of type html describing a red border and bold text for 'I'm melting!'"
which might have the same effect
whilst supporting arbitrary events to format themselves more flexibly
and still letting bots do custom formatting for a given type they know about
especially if we did the "extends m.severity" idea, and inherited the loglevel field from a generic mixin
tl;dr: what is special about notices here? can't we mixin fields from an m.severity event type to format up any given event for a given severity level?

My counterpoint is: Letting everybody come up with their own idea of the arbitrary HTML styling for 'this is an error' sounds like a recipe for technicolour vomit. Indeed, the entire point of this PR is to avoid every bot having its own *wording* for 'this is an error'.

**Comment by Half-Shot** [comment]
So the plan for this PR was to add metadata to a notice to give it in essence a priority and for the option for *any* client (HTML or not) to style it. As @richvdh says, the point for me was to avoid bots styling their messages at all.

You'll have to explain what you mean by:
> mixin fields from an m.severity event type to format up any given event for a given severity level?

It was my intention to hand spec levels so we don't end up with every bot coming up with it's own levles, or clients coming up with theirs. Having a strict set of levels, that are agreed upon gives client writers and bot writers something to aim at.

**Comment by turt2live** [comment]
It would be nice if there was an option for "no status" for the cases where the bot isn't reacting to user input, but instead responding to an outside event. 

A more specific use case: A bot that announces people entering a building (see image).
image

Although this message is informational, some clients may pick up on the status and decide to wrap it in a blue border that interrupts chat flow (which is not the intention of this particular event). An option to hint that clients should simply mute the text (ie: no status) would be ideal for these particular events.

*Apologies if I misinterpreted the change - it appears to me as though `status` becomes required*

**Comment by Half-Shot** [comment]
>Apologies if I misinterpreted the change - it appears to me as though status becomes required

Heh, no plans to make that required. The client should act as it currently does in the absence imo. The status is just for extra metadata.

**Comment by Half-Shot** [comment]
So uh, coming back to this a year later is there a reason why this can't be merged?

File: event-schemas/schema/m.room.message#m.notice [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
       body:
         description: The notice text to send.
         type: string
+      status:
+        description: A status hint to the client that represents the result of an operation or the priority of a message.
+        enum:
+          - info
+          - warning
+          - error
+          - critical
```
[the remark this answers is not in the document]
Comment (Half-Shot) [reply event-schemas/schema/m.room.message#m.notice]: Good shout, works for me.

**Comment by Half-Shot** [comment]
I need to really make this its own proposal now.

**Comment by Half-Shot** [comment]
I don't think this is likely to get merged, and will probably be re-evaluated after extensible events.
